hub/tests/test_files.py

```diff
@@ -0,0 +1,239 @@
+    assert __files == files, f"No file founded : {files}"
+
+
+def test_searchfile_list():
+    """
+    test searching the files if there were some filea when it takes the file name
+    파일이 여러 개 있을 때(파일명 지정)
+    """
+    # 정상적으로 작동해야 하는 결과
+    __files = [
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2\\test2mp4.mp4",
+    ]
+
+    # 실행시켰을 때 결과
+    directory = (
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata"
+    )
+    filename = "test2mp4.mp4"
+    recursive = True
+    extension = None
+    files = search_file(directory, filename, recursive, extension)
+
+    # 결과 비교
+    assert (
+        __files == files
+    ), f"Couldn't read all file list about {filename} on {directory}"
+
+
+def test_searchfile_all():
+    """
+    test returning the all files list (include subfolders' one)
+    폴더 안 모든 파일 리스트 출력(하위 폴더 포함)
+    """
+    # 정상적으로 작동해야 하는 결과
+    __files = [
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata3",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testmp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testtxt.txt",
+    ]
+
+    # 실행시켰을 때 결과
+    directory = (
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata"
+    )
+    filename = None
+    recursive = True
+    extension = None
+    files = search_file(directory, filename, recursive, extension)
+
+    # 결과 비교
+    assert (
+        __files == files
+    ), f"Couldn't read all file list on {directory} (include subfolders)"
+
+
+def test_searchfile_mp4():
+    """
+    test returning the all .mp4 files list (include subfolders' one)
+    폴더 안 모든 mp4 파일 리스트 출력(하위 폴더 포함)
+    """
+    # 정상적으로 작동해야 하는 결과
+    __files = [
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\test2mp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testmp4.mp4",
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata\\testdata2\\test2mp4.mp4",
+    ]
+
+    # 실행시켰을 때 결과
+    directory = (
+        "C:\\Users\\user\\Documents\\GitHub\\2021-spring-project\\hub\\tests\\testdata"
+    )
+    filename = None
+    recursive = True
+    extension = "mp4"
+    files = search_file(directory, filename, recursive, extension)
```

Comment:
parameter의 값으로 사용되는 1회성 코드들은 한줄로 합쳐도 좋을 것 같아요!
위에 작성된 내용들 모두 동일합니다!
```suggestion
    files = search_file(directory=`directory`, filename=`None`, recursive=`True`, extension=`"mp4"`)
```

Reply:
수정했습니다!